src/SFA.DAS.Roatp.ProviderModeration.Web/Controllers/ErrorController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Security.Claims;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Roatp.ProviderModeration.Web.Infrastructure;
+using SFA.DAS.Roatp.ProviderModeration.Web.Models;
+
+namespace SFA.DAS.Roatp.ProviderModeration.Web.Controllers;
+public class ErrorController : Controller
+{
+    private readonly ILogger<ErrorController> _logger;
+
+    public ErrorController(ILogger<ErrorController> logger)
+    {
+        _logger = logger;
+    }
+
+    [AllowAnonymous]
+    [Route("Error/{statuscode}")]
+    public IActionResult HttpStatusCodeHandler(int statusCode)
+    {
+        switch (statusCode)
+        {
+            case 403:
+            case 404:
+                return View("PageNotFound");
+            default:
+                ErrorViewModel errorViewModel = new()
+                {
+                    HomePageUrl = Url.RouteUrl(RouteNames.GetProviderDescription)!
+                };
+                return View("ErrorInService", errorViewModel);
+        }
+    }
+
+    [AllowAnonymous]
+    [Route("Error")]
+    public IActionResult ErrorInService()
+    {
+        var feature = HttpContext.Features.Get<IExceptionHandlerPathFeature>();
+        ErrorViewModel errorViewModel = new()
+        {
+            HomePageUrl = Url.RouteUrl(RouteNames.GetProviderDescription)!
+        };
+
+        if (User.Identity!.IsAuthenticated)
+        {
+            _logger.LogError(feature!.Error, "Unexpected error occured during request to path: {path} by user: {user}", feature.Path, HttpContext.User.FindFirstValue(ClaimTypes.Upn));
```

Comment:
is it usual for us to log the user so explicitly?  I would just log the upn or something similarly anonymous

Reply:
Yes, I made the same changes in the Apprentice and Employer web